File: game-server/src/main/java/org/forfun/mmorpg/game/scene/actor/Monster.java
package org.forfun.mmorpg.game.scene.actor;

public class Monster extends Creature {
	
	private int modelId;
	
	private String name;
	

	public Monster() {
	}


	public int getModelId() {
		return modelId;
	}


	public void setModelId(int modelId) {
		this.modelId = modelId;
	}


	public String getName() {
		return name;
	}


	public void setName(String name) {
		this.name = name;
	}
	
	@Override
	public ActorType getType() {
		return ActorType.Monster;
	}
	
}

File: game-server/src/main/java/org/forfun/mmorpg/game/scene/actor/Monster.kt
package org.forfun.mmorpg.game.scene.actor

import org.forfun.mmmorpg.game.scene.actor.Creature

open class Monster : Creature() {

    var modelId: Int = 0

    var name: String? = null

    override val type: ActorType
        get() = ActorType.Monster

    override fun getId(): Long {
        return 0
    }
}
